floris/simulation/flow_field.py

```diff
@@ -69,6 +76,40 @@ def wind_directions_validator(self, instance: attrs.Attribute, value: NDArrayFlo
         """Using the validator method to keep the `n_wind_directions` attribute up to date."""
         self.n_wind_directions = value.size
 
+    @het_config.validator
+    def het_config_validator(self, instance: attrs.Attribute, value: dict | None) -> None:
+        """Using the validator method to check that the het_config dictionary has the correct
+        key-value pairs.
+        """
+        if value is not None:
+            # Check that the correct keys are supplied for the het_config dict
+            for k in ["speed_ups", "x_locs", "y_locs"]:
+                if k not in value.keys():
+                    raise ValueError(
+                        "het_config must contain entries for 'speeds_ups', 'x_locs', and 'y_locs',"
+                        f" with 'z_locs' as optional. Missing '{k}'."
+                    )
+            if "z_locs" not in value:
+                # If only a 2D case, add "None" for the z locations
+                value["z_locs"] = None
+
+    @het_map.validator
+    def het_map_validator(self, instance: attrs.Attribute, value: list | None) -> None:
```

Comment:
Same as above

Reply:
Same comment. Maybe since this isn't a passed variable you could use the following in the attribute definition, which creates a new list as the default value. However, with `init=False`, and a user not providing an actual value to this, then you'd need to check that it's not an empty list, which I think is preferable.
```python
het_map: list = field(init=False, factory=list, on_setattr=attrs.setters.validate)

@het_map.validator
def het_map_validator(self, instance, value):
    if value == []:
        return
    if self.n_wind_directions!= np.array(value).shape[0]:
        raise ValueError(
            "The het_map's wind direction dimension not equal to number of wind directions."
        )
```